gcloud/monitoring/query.py

```diff
@@ -0,0 +1,661 @@
+
+        self._client = client
+        self._end_time = end_time
+        self._start_time = start_time
+        self._filter = _Filter(metric_type)
+
+        self._per_series_aligner = None
+        self._alignment_period_seconds = None
+        self._cross_series_reducer = None
+        self._group_by_fields = ()
+
+    def __iter__(self):
+        return self.iter()
+
+    @property
+    def metric_type(self):
+        """The metric type name."""
+        return self._filter.metric_type
+
+    @property
+    def filter(self):
+        """The filter string.
+
+        This is constructed from the metric type, the resource type, and
+        selectors for the group ID, monitored projects, resource labels,
+        and metric labels.
+        """
+        return str(self._filter)
+
+    def select_interval(self, end_time, start_time=None):
+        """Copy the query and set the query time interval.
+
+        Example::
+
+            import datetime
+
+            now = datetime.datetime.utcnow()
+            query = query.select_interval(
+                end_time=now,
+                start_time=now - datetime.timedelta(minutes=5))
+
+        As a convenience, you can alternatively specify the end time and
+        an interval duration when you create the query initially.
+
+        :type end_time: :class:`datetime.datetime`
+        :param end_time: The end time (inclusive) of the time interval
+            for which results should be returned, as a datetime object.
+
+        :type start_time: :class:`datetime.datetime` or None
+        :param start_time: The start time (exclusive) of the time interval
+            for which results should be returned, as a datetime object.
+            If not specified, the interval is a point in time.
+
+        :rtype: :class:`Query`
+        :returns: The new query object.
+        """
+        new_query = self.copy()
+        new_query._end_time = end_time
+        new_query._start_time = start_time
+        return new_query
+
+    def select_group(self, group_id):
+        """Copy the query and add filtering by group.
+
+        Example::
+
+            query = query.select_group('1234567')
+
+        :type group_id: string
+        :param group_id: The ID of a group to filter by.
+
+        :rtype: :class:`Query`
+        :returns: The new query object.
+        """
+        new_query = self.copy()
+        new_query._filter.group_id = group_id
+        return new_query
+
+    def select_projects(self, *args):
+        """Copy the query and add filtering by monitored projects.
+
+        This is only useful if the target project represents a Stackdriver
+        account containing the specified monitored projects.
+
+        Examples::
+
+            query = query.select_projects('project-1')
+            query = query.select_projects('project-1', 'project-2')
+
+        :param args: Project IDs limiting the resources to be included
+            in the query.
```

Comment:
I typically would align `in the query` with `Project IDs ... `, but it's not necessary.